src/components/comments/canvas-pin-adapter/index.ts

```diff
@@ -179,6 +183,34 @@ export class CanvasPin implements PinAdapter {
     return mouseElement;
   }
 
+  /**
+   * @function resetPins
+   * @description Unselects selected pin and removes temporary pin.
+   * @param {that} this - The canvas pin adapter instance.
+   * @param {KeyboardEvent} event - The keyboard event object.
+   * @returns {void}
+   * */
+  private resetSelectedPin(): void {
+    if (!this.selectedPin) return;
+    this.selectedPin.removeAttribute('active');
+    this.selectedPin = null;
+  }
+
+  /**
+   * @function resetPins
+   * @description Unselects selected pin and removes temporary pin.
+   * @param {that} this - The canvas pin adapter instance.
+   * @param {KeyboardEvent} event - The keyboard event object.
+   * @returns {void}
+   * */
+  private resetPins(event?: KeyboardEvent, that: this = this): void {
+    if (event && event?.key !== 'Escape') return;
+
+    // use of "that" because using "this" normally wasn't working when pressing escape
+    that.resetSelectedPin();
+    that.removeAnnotationPin('temporary-pin');
+  }
```

Comment:
You can use a function expression here or bind this method into context using bind. 

And its should be together all callbacks, at the bottom of the class. Just a convention :)

```suggestion
  private resetPins = (event?: KeyboardEvent): void => {
    if (event && event?.key !== 'Escape') return;

    this.resetSelectedPin();
    this.removeAnnotationPin('temporary-pin');
  }
```